subsys/bluetooth/controller/util/mfifo.h

```diff
@@ -0,0 +1,177 @@
+
```

Comment:
This file needs comments on each function explaining what they do.

Reply:
I've documented this in https://github.com/zephyrproject-rtos/zephyr/pull/12167

Reply:
Note that  #12167 was spun out of https://github.com/NordicPlayground/zephyr-bt/pull/26 as that was the intersection between non-split and split stack.

Reply:
Thank you @mped-oticon, that is a great effort. A while back I added the `memq_link_t` in order to make the code more readable, and with your comments this closes the circle.

Reply:
>I've documented this in #12167

That PR doesn't include `mfifo` but you have another branch for that.

Reply:
Remind me why we need a different fifo/list implementation?

Reply:
For 3 reasons:

1) Memory efficiency and code size
2) Latency: this is lockless unlike the kernel's FIFOs, never disables interrupts
3) Independence from the kernel. This is only used on interrupt context, should never block or use kernel primitives